src/main/kotlin/hr/foi/pop/backend/services/UserService.kt

```diff
@@ -41,6 +44,26 @@ class UserService {
         return user
     }
 
+    @Throws(ActivateUserException::class)
+    fun activateOrDeactivateUser(userId: String, activateUserDefinition: ActivateUserDefinitions): User {
+        try {
+            val user: User = userRepository.getReferenceById(userId.toInt())
```

Comment:
This `toInt()` I already commented on in UserController's comment. This method should've never received a string for an ID in the first place. It is not its responsibility to parse strings to integers, but rather to deal with users.
After that's done, the awful try-catch block is going away and we'll get to enjoy a beautiful, clean UserService's method purely made for handling activation status changes.

Reply:
Uncle Mislav in action.